src/instrumentsscene/qml/MuseScore/InstrumentsScene/internal/InstrumentsOnScoreView.qml

```diff
@@ -26,9 +26,11 @@ import QtQuick.Controls 2.12
 import Muse.Ui 1.0
 import Muse.UiComponents 1.0
 import MuseScore.InstrumentsScene 1.0
+import MuseScore.Project 1.0
 
 Item {
     id: root
+    required property InstrumentsOnScoreListModel instrumentsOnScoreModel
 
     readonly property bool hasInstruments: instrumentsOnScoreModel.count > 0
     readonly property alias isMovingUpAvailable: instrumentsOnScoreModel.isMovingUpAvailable
```

Comment:
The errors you are seeing are caused by these two aliases:
```
13:08:15.122 | DEBUG | main_thread     | createDialog    | [qml] failed create component: ../../MuseScore/Project/NewScoreDialog.qml, err: qrc:/qml/MuseScore/Project/NewScoreDialog.qml:87 Type ChooseInstrumentsAndTemplatesPage unavailable
qrc:/qml/MuseScore/Project/internal/NewScore/ChooseInstrumentsAndTemplatesPage.qml:174 Type ChooseInstrumentsPage unavailable
qrc:/qml/MuseScore/InstrumentsScene/ChooseInstrumentsPage.qml:196 Type InstrumentsOnScoreView unavailable
qrc:/qml/MuseScore/InstrumentsScene/internal/InstrumentsOnScoreView.qml:36 Invalid alias reference. Unable to find id "instrumentsOnScoreModel"
```
Indeed, `instrumentsOnScoreModel` is not an id anymore in this context, but a property. That means that it cannot be used for `alias` properties. 
Fortunately, the solution is easy: replace `alias` with `bool`.
But there is also another solution, that may be even better. If you look where `isMovingUpAvailable` and `isMovingDownAvailable` are used, you see that they are only used in `ChooseInstrumentsPage.qml`. But now that file has direct access to the `instrumentsOnScoreModel`, so there is no need anymore to take these properties via `InstrumentsOnScoreView`. 
So, the solution I would recommend is:
- remove these two alias properties from InstrumentsOnScoreView
- In ChooseInstrumentsPage, replace `instrumentsOnScoreView.isMovingUpAvailable` with `root.instrumentsOnScoreModel.isMovingUpAvailable`, and similar for `isMovingDownAvailable`.

While at it, you'll probably notice that InstrumentsOnScoreView contains a lot of functions (`instruments()`, `moveSelectedInstrumentsDown()`, etc) that are only for the purpose of exposing information of `instrumentsOnScoreModel`. But again, that is not necessary anymore, because now `instrumentsOnScoreModel` is available directly. So, it would be great to remove those functions, and use `instrumentsOnScoreModel` directly in ChooseInstrumentsPage.

Reply:
It is kind of frustrating for me that after so much of  instructions I could not find out the correct solution  but I am listing my progress.  @cbjeukendrup  As per your instructions I update the InstrumentsDialog.qml, ChooseInstrumentsPage.qml and InstrumentsOnScoreView.qml. And code of those files are as follows:
InstrumentsDialog.qml
```qml
StyledDialogView {
    ......
    InstrumentsOnScoreListModel {
        id: instrumentsOnScoreModel
    }
    ......
}
```
ChooseInstrumentsPage.qml
```qml
function instruments() {
        if (root.canSelectMultipleInstruments) {
            return root.instrumentsOnScoreModel.instruments() // changed code
        }

        return [ instrumentsModel.selectedInstrument ]
}
function currentOrder() {
        if (root.canSelectMultipleInstruments) {
            return root.instrumentsOnScoreModel.currentOrder() // changed code
        }

        return undefined
}
function addSelectedInstrumentsToScore() {
        root.instrumentsOnScoreModel.addInstruments(instrumentsModel.selectedInstrumentIdList()) // changed code

        Qt.callLater(root.instrumentsOnScoreModel.scrollViewToEnd) // changed code
}
FlatButton {
        enabled: root.instrumentsOnScoreModel.isMovingUpAvailable // changed code
}
FlatButton {
        enabled: root.instrumentsOnScoreModel.isMovingDownAvailable // changed code
}
```
And I deleted two alice properties isMovingUpAvailable, isMovingDownAvailable and all those functions instruments(), currentOrder(), addInstruments, moveSelectedInstrumentsUp(), moveSelectedInstrumentsDown(), scrollViewToEnd() from the InstrumentsOnScoreView.qml.
After this Newscore creation dialog is opening but i can not add instruments to the list and It is showing in application output that 
error: qrc:/qml/MuseScore/InstrumentsScene/ChooseInstrumentsPage.qml:85: TypeError: Cannot read property 'addInstruments' of null

Reply:
Yes Yes, it is there in the InstrumentsOnScoreView.qml ,I mistakenly add in the comment.